src/wp-admin/includes/misc.php

```diff
@@ -1641,3 +1641,133 @@ function wp_check_php_version() {
 
 	return $response;
 }
+
+/**
+ * Creates and returns the markup for an admin notice.
+ *
+ * @since 6.3.0
+ *
+ * @param string $message The message.
+ * @param array  $args {
+ *     Optional. An array of arguments for the admin notice. Default empty array.
+ *
+ *     @type string   $type               Optional. The type of admin notice.
+ *                                        For example, 'error', 'success', 'warning', 'info'.
+ *                                        Default empty string.
+ *     @type bool     $dismissible        Optional. Whether the admin notice is dismissible. Default false.
+ *     @type string   $id                 Optional. The value of the admin notice's ID attribute. Default empty string.
+ *     @type string[] $additional_classes Optional. A string array of class names. Default empty array.
+ *     @type bool     $paragraph_wrap     Optional. Whether to wrap the message in paragraph tags. Default true.
+ * }
+ */
+function wp_get_admin_notice( $message, $args ) {
+	$defaults = array(
+		'type'               => '',
+		'dismissible'        => false,
+		'id'                 => '',
+		'additional_classes' => array(),
+		'paragraph_wrap'     => true,
+	);
+
+	$args = wp_parse_args( $args, $defaults );
+
+	/**
+	 * Filters the arguments for an admin notice.
+	 *
+	 * @since 6.3.0
+	 *
+	 * @param array  $args    The arguments for the admin notice.
+	 * @param string $message The message for the admin notice.
+	 */
+	$args    = apply_filters( 'wp_admin_notice_args', $args, $message );
+	$id      = '';
+	$classes = 'notice';
+
+	if ( is_string( $args['id'] ) ) {
+		$trimmed_id = trim( $args['id'] );
+
+		if ( '' !== $trimmed_id ) {
+			$id = 'id="' . esc_attr( $trimmed_id ) . '" ';
+		}
+	}
+
+	if ( is_string( $args['type'] ) ) {
+		$type = trim( $args['type'] );
+
+		if ( '' !== $type ) {
+			if ( 'updated' === $type ) {
+				$type = 'success';
+			}
+
+			if (
+				// Don't add "notice-" when it already exists.
+				str_starts_with( $type, 'notice-' ) ||
+				// Don't add "notice-" for a type containing spaces.
+				str_contains( $type, ' ' ) ||
+				// Don't add "notice-" for unknown types.
+				! in_array( $type, array( 'error', 'success', 'warning', 'info' ), true )
```

Comment:
Also thinking of removing this for the same reason mentioned in [this comment](https://github.com/WordPress/wordpress-develop/pull/4119/files#r1272537362). I don't think there's a need to limit the number of accepted notice types. If someone doesn't want a `notice-` prefix, they can use `additional_classes` instead.

Reply:
I agree. This is a new function, so (for a change) we don't need to worry about backwards compatibility, so let's not add multiple pathways to reach the same end result if we can avoid it.

Reply:
Thanks Joe! What do you think about the other two conditions here?

The first prevents a double `notice-` prefix. Without this, if someone passes `notice-success`, they'll get `notice-notice-success`, but I don't really know why they would do that and expect anything different. The parameter is `$type`, not `$class_name`, and the documentation provides examples that don't contain the `notice-` prefix. I think this may be another candidate for removal.

For the other condition, I'm not sure how (or whether) we should handle a `$type` that contains a string. At the moment, the space-separated value is just appended to `$classes`, but I'd like to try to keep the parameters strict enough that users don't get stuck too much in HTML mode and try to just pass a space-separated string of classes to `$type` instead of using `$additional_classes` as intended.

We could do `explode( ' ', trim( $args['type'] ) )[0]`, for example, and just use the first "word" in the space-separated string, or we could use `_doing_it_wrong()`, which is probably the better thing to do. Thoughts?

Reply:
I'm so accustomed to getting notifications in trac, that I don't go looking for updates on tickets. Sorry I missed this!

I agree that nobody should be doing that, and I'm not sure we need to be protecting people against mistakes like that. As you say, it's `$type`, not `$class_name`; there's already someplace to add `$additional_classes`, the error should be immediately apparent. And, if somebody *wanted* `notice-notice-success`, this makes it harder to achieve. Even if that's silly.

I'm in favor of `_doing_it_wrong()` if $type is more than a simple string. I don't think we should be changing values, as that creates unexplained results - devs sending 'notice repaired' and only getting 'notice' with no explanation. `_doing_it_wrong()` explains to the dev why there's a problem.

Reply:
Yes, the more I look at this, the more I feel like it's overdoing it to try and control the usage of the `$type` attribute. I think it's more predictable if the code always generates a class of the shape `notice-$type`. All of these variations just make the code more obscure. If we were compensating for pre-existing code paths that expected different values, I'd feel differently.

Reply:
Updated in [05def7e](https://github.com/WordPress/wordpress-develop/pull/4119/commits/05def7e6643c1303cb5045d9ff739d29570f1764).